activerecord/lib/active_record/type/integer.rb

```diff
@@ -38,18 +38,21 @@ def cast_value(value)
 
       def ensure_in_range(value)
         unless range.cover?(value)
-          raise RangeError, "#{value} is out of range for #{self.class} with limit #{limit || 4}"
+          raise RangeError, "#{value} is out of range for #{self.class} with limit #{limit_or_default}"
         end
       end
 
       def max_value
-        limit = self.limit || 4
-        1 << (limit * 8 - 1) # 8 bits per byte with one bit for sign
+        1 << (limit_or_default * 8 - 1) # 8 bits per byte with one bit for sign
       end
 
       def min_value
         -max_value
       end
+
+      def limit_or_default
+        limit || 4
```

Comment:
here magic number `4` can be extracted to a named variable/constant. Or the method can be better named to take care of that.